### DISCUSSION TITLE
How to properly handle URL params in prisma query?

### DISCUSSION THREAD

**Discussion by angelhodar** [prisma/prisma#10358] [Q&A] [answered]
Hi! I have been using mongoose in my NodeJS backend and I would like to try prisma. The thing is that im using the package api-query-params which transforms the req.query in a specific mongoose query to handle some aspects of the data for me (pagination, skip, limit, join, filters, etc). Is there an official package from prisma to handle that? What is the proper way to handle those common api CRUD params with prisma?

**Comment by austincrim** [comment, marked as answer]
Hi @angelhodar 👋!

We don't have an official package to do this. Depending on your backend setup, using query params in a Prisma query should be relatively straightforward. Query options like `skip` and `take` just require that you pass in an integer.

If you have access to `req.query`, you would need to make sure to parse the values into a `Number` and then pass them into your Prisma query like normal.

Here's an example using Express:
```ts
const express = require('express')
const { PrismaClient } = require('@prisma/client')
const app = express()
const prisma = new PrismaClient()

app.get('/', (req, res) => {
  const { offset, take } = req.query  // pull the string values off the parsed query string
  const result = await prisma.post.findMany({
    offset: Number(offset),  // cast the value to `Number` before passing it into the query
    take: Number(take)
  })
  res.send(result)
})

app.listen(3000)
```

If you provide me more details about your backend, I would be happy to give a more specific example, otherwise, let me know if you have more questions!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
